datasets/ru/egrul/crawler.py

```diff
@@ -446,6 +447,52 @@ def parse_address(context: Context, entity: Entity, el: Element) -> None:
     entity.add("address", address)
 
 
+def compile_abbreviations(context) -> List[Tuple[str, re.Pattern, List[str]]]:
+    """
+    Load abbreviations and compile regex patterns from the YAML config.
+    Returns:
+    List[Tuple[str, re.Pattern, List[str]]]: A list of tuples containing canonical abbreviations
+    and their compiled regex patterns for substitution.
+    """
+    yaml = context.dataset.config.get("types")
+    abbreviations = []
+    for canonical, phrases in yaml.items():
+        # we want to match the whole word and allow for ["] or ['] at the beginning
+        for phrase in phrases:
+            phrase_pattern = rf"^[ \"']?{re.escape(phrase)}"
+
+            compiled_pattern = re.compile(phrase_pattern, re.IGNORECASE)
+            # Append the canonical form, compiled regex pattern, and sorted phrases to the list
+            abbreviations.append((canonical, compiled_pattern, phrase))
+
+    abbreviations.sort(key=lambda x: len(x[2]), reverse=True)
+    return abbreviations
+
+
+def substitute_abbreviations(
+    name: str, abbreviations: List[Tuple[str, re.Pattern, List[str]]]
```

Comment:
we could declare a [type alias](https://docs.python.org/3/library/typing.html#type-aliases) AbbreviationList and then use that in the type hints to the functions where we compile and use them